domain-models-runtime/src/main/java/org/folio/rest/persist/PostgresClient.java

```diff
@@ -2801,31 +2802,92 @@ public void selectSingle(AsyncResult<SQLConnection> conn, String sql, Handler<As
    * @param replyHandler  The query result or the failure.
    */
   public void selectSingle(String sql, JsonArray params, Handler<AsyncResult<JsonArray>> replyHandler) {
-    client.getConnection(conn -> selectSingle(conn, sql, params, closeAndHandleResult(conn, replyHandler)));
+    getSQLConnection(conn -> selectSingle(conn, sql, params, closeAndHandleResult(conn, replyHandler)));
+  }
+
+  private static String legacySql(String sql, JsonArray params) {
+    if (params.isEmpty()) {
+      return sql;
+    }
+    StringBuilder preparedSql = new StringBuilder();
+    int pos = 1;
+    for (int i = 0; i < sql.length(); i++) {
+      if (sql.charAt(i) =='?') {
+        preparedSql.append("$" + pos);
+        pos++;
+      } else {
+        preparedSql.append(sql.charAt(i));
+      }
+    }
+    return preparedSql.toString();
+  }
```

Comment:
Unmasking `??`, including the params.isEmpty() case, is missing. Better:

```
  /**
   * Replace each "??" by "?" and each other "?" by "$1", "$2", ...
   *
   * The old vertx-mysql-postgresql-client-jasync driver uses double question mark
   * to mask any non-placeholder question mark:
   * <a href="https://github.com/jasync-sql/jasync-sql/blob/1.0.17/postgresql-async/src/main/java/com/github/jasync/sql/db/postgresql/PreparedStatementHolder.kt#L23">
   * jasync-sql PreparedStatementHolder</a>
   */
  private static String legacySql(String sql) {
    StringBuilder preparedSql = new StringBuilder();
    int pos = 1;
    boolean masked = false;
    for (int i = 0; i < sql.length(); i++) {
      if (sql.charAt(i) != '?' || masked) {
        preparedSql.append(sql.charAt(i));
        masked = false;
      } else if (i + 1 < sql.length() && sql.charAt(i + 1) == '?') {
        masked = true;
      } else {
        preparedSql.append('$').append(pos);
        pos++;
      }
    }
    return preparedSql.toString();
  }
```

Reply:
```
  @Test
  public void selectParamQuestionMarkJsonb(TestContext context) {
    postgresClient()
    // "jsonb ? field" checks whether field is a key in jsonb.
    // double the question mark to mask from parameter substitution
    .select("SELECT '{\"field\": \"value\"}'::jsonb ?? ?",
        new JsonArray().add("field"), context.asyncAssertSuccess(select -> {
          context.assertEquals(true, select.iterator().next().getBoolean(0));
        }));
  }
```
```
  @Test
  public void selectParamQuestionMarkRegexp(TestContext context) {
    postgresClient()
    // ? in the regexp makes the space optional.
    // double the question mark to mask from parameter substitution
    .select("SELECT 'abc 123' ~ '^[a-z]+ ??[0-9]+$' AND 'xyz' = ?",
        new JsonArray().add("xyz"), context.asyncAssertSuccess(select -> {
          context.assertEquals(true, select.iterator().next().getBoolean(0));
        }));
  }
```

Reply:
OK. But we'll just not do this and require people to use the new syntax, I suppose.

Reply:
Yes.